cmd/boost/wallet_cmd.go

```diff
@@ -0,0 +1,476 @@
+	Subcommands: []*cli.Command{
+		walletNew,
+		walletList,
+		walletBalance,
+		walletExport,
+		walletImport,
+		walletGetDefault,
+		walletSetDefault,
+		walletDelete,
+	},
+}
+
+var walletNew = &cli.Command{
+	Name:      "new",
+	Usage:     "Generate a new key of the given type",
+	ArgsUsage: "[bls|secp256k1 (default secp256k1)]",
+	Action: func(cctx *cli.Context) error {
+		ctx := lcli.ReqContext(cctx)
+
+		sdir, err := homedir.Expand(cctx.String("repo"))
+		if err != nil {
+			return err
+		}
+
+		n, err := node.Setup(ctx, sdir)
+		if err != nil {
+			return err
+		}
+
+		t := cctx.Args().First()
+		if t == "" {
+			t = "secp256k1"
+		}
+
+		nk, err := n.Wallet.WalletNew(ctx, types.KeyType(t))
+		if err != nil {
+			return err
+		}
+
+		fmt.Println(nk.String())
+
+		return nil
+	},
+}
+
+var walletList = &cli.Command{
+	Name:  "list",
+	Usage: "List wallet address",
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name:    "addr-only",
+			Usage:   "Only print addresses",
+			Aliases: []string{"a"},
+		},
+		&cli.BoolFlag{
+			Name:    "id",
+			Usage:   "Output ID addresses",
+			Aliases: []string{"i"},
+		},
+		&cli.BoolFlag{
+			Name:    "market",
+			Usage:   "Output market balances",
+			Aliases: []string{"m"},
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		ctx := lcli.ReqContext(cctx)
+
+		sdir, err := homedir.Expand(cctx.String("repo"))
+		if err != nil {
+			return err
+		}
+
+		n, err := node.Setup(ctx, sdir)
+		if err != nil {
+			return err
+		}
+
+		api, closer, err := lcli.GetGatewayAPI(cctx)
+		if err != nil {
+			return fmt.Errorf("cant setup gateway connection: %w", err)
+		}
+		defer closer()
+
+		afmt := NewAppFmt(cctx.App)
+
+		addrs, err := n.Wallet.WalletList(ctx)
+		if err != nil {
+			return err
+		}
+
+		// Assume an error means no default key is set
+		def, _ := n.Wallet.GetDefault()
+
+		tw := tablewriter.New(
+			tablewriter.Col("Address"),
+			tablewriter.Col("ID"),
+			tablewriter.Col("Balance"),
+			tablewriter.Col("Market(Avail)"),
+			tablewriter.Col("Market(Locked)"),
+			tablewriter.Col("Nonce"),
+			tablewriter.Col("Default"),
+			tablewriter.NewLineCol("Error"))
+
+		for _, addr := range addrs {
+			if cctx.Bool("addr-only") {
+				afmt.Println(addr.String())
+			} else {
+				a, err := api.StateGetActor(ctx, addr, types.EmptyTSK)
+				if err != nil {
+					if !strings.Contains(err.Error(), "actor not found") {
+						tw.Write(map[string]interface{}{
+							"Address": addr,
+							"Error":   err,
+						})
+						continue
+					}
+
+					a = &types.Actor{
+						Balance: big.Zero(),
+					}
+				}
+
+				row := map[string]interface{}{
+					"Address": addr,
+					"Balance": types.FIL(a.Balance),
+					"Nonce":   a.Nonce,
+				}
+				if addr == def {
+					row["Default"] = "X"
+				}
+
+				if cctx.Bool("id") {
+					id, err := api.StateLookupID(ctx, addr, types.EmptyTSK)
+					if err != nil {
+						row["ID"] = "n/a"
+					} else {
+						row["ID"] = id
+					}
+				}
+
+				if cctx.Bool("market") {
```

Comment:
This is the same as Lotus, but I think we could get rid of this argument, and always display the Market balances... WDYT?

I'd say a user cares about ALL their balances across a given wallet, or they care only about the addresses in which case they'd use `--addr-only=true`.

Reply:
Yeah, I would prefer to keep it minimal intervention. Always display market balance is better.

Reply:
I think we should always display a market balance 👍 
I also agree it would be nice to be able to show wallets even when not connected to the internet, so let's also add the `--addr-only=true` flag as you suggested.

Reply:
Removed `market` flag.